src/client.rs

```diff
@@ -403,44 +457,66 @@ fn apply_despawns(
 fn apply_update_components(
     cursor: &mut Cursor<&[u8]>,
     world: &mut World,
+    state: &mut ReceiveState,
     entity_map: &mut ServerEntityMap,
     entity_ticks: &mut ServerEntityTicks,
     mut stats: Option<&mut ClientStats>,
+    command_markers: &CommandMarkers,
     replication_fns: &ReplicationFns,
     message_tick: RepliconTick,
 ) -> bincode::Result<()> {
     let message_end = cursor.get_ref().len() as u64;
     while cursor.position() < message_end {
-        let entity = deserialize_entity(cursor)?;
+        let server_entity = deserialize_entity(cursor)?;
         let data_size: u16 = bincode::deserialize_from(&mut *cursor)?;
-        let Some(mut entity) = entity_map.get_by_server(world, entity) else {
+
+        let Some(client_entity) = entity_map.get_by_server(server_entity) else {
             // Update could arrive after a despawn from init message.
-            debug!("ignoring update received for unknown server's {entity:?}");
+            debug!("ignoring update received for unknown server's {server_entity:?}");
             cursor.set_position(cursor.position() + data_size as u64);
             continue;
         };
         let entity_tick = entity_ticks
-            .get_mut(&entity.id())
+            .get_mut(&client_entity)
             .expect("all entities from update should have assigned ticks");
         if message_tick <= *entity_tick {
-            trace!("ignoring outdated update for client's {:?}", entity.id());
+            trace!("ignoring outdated update for client's {client_entity:?}");
             cursor.set_position(cursor.position() + data_size as u64);
             continue;
         }
         *entity_tick = message_tick;
 
+        let (mut entity_markers, mut commands, mut query) = state.get_mut(world);
+        let mut client_entity = query.get_mut(client_entity).unwrap();
```

Comment:
We have test for combining insertions or updates with despawns, they pass.

But yes, if not server, but client itself despawn an entity in `PreUpdate`, it will panic. This is how it was before (`World::entity_mut` panics if there is no such entity).
I probably should replace it with `expect` with something like "replicated entities can be despawned only by server"?